src/main/java/frc/robot/commands/AutoBalance.java

```diff
@@ -41,46 +38,71 @@ public AutoBalance(SwerveDrivetrain drive) {
   @Override
   public void initialize() {
     m_balanceController.setSetpoint(AutoConstants.DESIRED_BALANCE_ANGLE);
+    m_isLevel = false;
+
+    m_timer.reset();
+    m_timer.stop();
+
+    m_balanceTimer.reset();
+    m_balanceTimer.stop();
   }
 
   // Called every time the scheduler runs while the command is scheduled.
   @Override
   public void execute() {
     // Gets how far the robot is tilted and calculates proper drive power from it
-    m_currentAngle = m_drive.getGyroPitch().getDegrees();
-    m_error = AutoConstants.DESIRED_BALANCE_ANGLE - m_currentAngle;
+    double m_currentAngle = m_drive.getGyroPitch().getDegrees();
+    double m_error = AutoConstants.DESIRED_BALANCE_ANGLE - m_currentAngle;
+    double m_drivePower = Math.min(m_balanceController.calculate(m_currentAngle), 1);
 
-    m_drivePower = Math.min(m_balanceController.calculate(m_currentAngle), 1);
     // Limit max power
-    m_drivePower = MathUtil.clamp(m_drivePower, -0.6, 0.6);
+    m_drivePower = MathUtil.clamp(m_drivePower, -0.5, 0.5);
 
-    double rollRateBalance = m_drive.getGyroRollPitch();
+    double rollRateBalance = m_drive.getGyroPitchRate();
```

Comment:
The number returned here is different now that it is adjusted for time elapsed.

Reply:
wouldn't it still be the rate of the roll, just properly calculated now as we're accouting for time elapsed?

Reply:
Correct. Just making a note for later.